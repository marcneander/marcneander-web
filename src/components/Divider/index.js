import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const propTypes = {
    width: PropTypes.number,
    inverse: PropTypes.bool
};

const defaultProps = {
    width: 20,
    inverse: false
};

const StyledDivider = styled.span`
    display: inline-block;
    height: 2px;
    margin: 30px 15px;
    background: ${props => (props.inverse ? '#fdf6e3' : '#293440')};
`;

const Divider = props => {
    const { width, inverse } = props;

    return <StyledDivider style={{ width }} inverse={inverse} />;
};

Divider.propTypes = propTypes;
Divider.defaultProps = defaultProps;

export default Divider;
