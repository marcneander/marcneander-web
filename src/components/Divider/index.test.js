/* global it, expect, describe */
import React from 'react';
import { render } from 'enzyme';
import Divider from './index';

describe('<Divider>', () => {
    it('should render correctly', () => {
        const wrapper = render(<Divider width={30} />);

        expect(wrapper).toMatchSnapshot();
    });
});
