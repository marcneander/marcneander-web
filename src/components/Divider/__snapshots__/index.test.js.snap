// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider> should render correctly 1`] = `
<span
  class="sc-bdVaJa eOeOIj"
  style="width:30px"
/>
`;
